### PR TITLE
[Translations] Merge Admin & Shared translations into single UI 

### LINKED ISSUE #6568: [Translations] Domain Support + Context Information + Merging Admin & Shared ...
- [x]  Merging Admin and Shared Translations
- [x]  Manage them within one single UI
- Shortcuts in the UI for frontend and backend locales Moved to new issue https://github.com/pimcore/pimcore/issues/11065
- [x] Shortcuts for domain filtering
- [x] Support for multiple domains (default `messages` and `admin`)

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves partially #6568

## Additional info  
- Merge Admin & Shared translations within one single UI
- Shortcuts for domain filtering

### CODE REVIEW

File: bundles/AdminBundle/Controller/Admin/TranslationController.php [hunk 1]
```diff
@@ -1273,4 +1269,26 @@ public function getWebsiteTranslationLanguagesAction(Request $request)
...
+     */
+    public function getTranslationDomainsAction(Request $request)
+    {
+        $translation = new Translation();
+
+        $domains = array_map(function ($domain) {
+            return [
+                'name' => $domain,
+            ];
+        }, $translation->getDao()->getAvailableDomains());
```
Comment (jdreesen) [inline bundles/AdminBundle/Controller/Admin/TranslationController.php]:
```suggestion
        $domains = array_map(
            fn ($domain) => ['name' => $domain],
            $translation->getDao()->getAvailableDomains(),
        );
```

File: models/Translation/Dao.php [hunk 2]
```diff
@@ -115,6 +115,23 @@ public function getAvailableLanguages()
...
+     *
+     * @return array
+     */
+    public function getAvailableDomains()
+    {
+        $domainTables = $this->db->fetchAll("SHOW TABLES LIKE 'translations_%'");
+        $domains = [];
+
+        foreach ($domainTables as $domainTable) {
+            $domains[] = str_replace('translations_', '', current($domainTable));
```
Comment (jdreesen) [inline models/Translation/Dao.php]:
```suggestion
            $domains[] = str_replace('translations_', '', $domainTable);
```

Comment (dvesh3) [reply models/Translation/Dao.php]: we just need value here:
image

Comment (jdreesen) [reply models/Translation/Dao.php]: Oh, wow! Didn't expect this!

Maybe writing it as follows would make it more clear:

```suggestion
            $domains[] = str_replace('translations_', '', $domainTable[array_key_first($domainTable)]);
```

**Review by mcop1 — COMMENTED** [review]
@dvesh3  I get an empty entry in the domain dropdown, when I

1.) Export the translations to a local csv file
2.) Make some changes to the translations
3.) Import the csv file again

**Comment by dvesh3** [comment]
> @dvesh3 I get an empty entry in the domain dropdown, when I
> 
> 1.) Export the translations to a local csv file 2.) Make some changes to the translations 3.) Import the csv file again

Fixed by setting default domain "messages" when domain is empty while creating/updating translations.

File: models/Translation/Dao.php [hunk 3]
```diff
@@ -15,6 +15,7 @@
 
 namespace Pimcore\Model\Translation;
 
+use mysql_xdevapi\Exception;
```
Comment (dvesh3) [inline models/Translation/Dao.php]:
```suggestion
```
unrelated

**Review by mcop1 — APPROVED** [review]
